### DISCUSSION TITLE
Error handling in Spring and HTMX

### DISCUSSION THREAD

**Discussion by martinbeentjes** [wimdeblauwe/wimdeblauwe.com#4] [Q&A]
How can I have an `@ControllerAdvice` for exceptions that could be both full page errors and htmx errors. For the latter, e.g., I have some POSTs that should happen in some cases, but are not always allowed. Then I'd like a nice modal, instead of the full page error page.

The `@ControllerAdvice` handles BadRequest Exceptions and returns an `error/400.html` template. However, I am also having some bad requests that should be for example be a modal view. 

Is it common to check for the HX-Request header in the controller advice and return the proper error fragment there? I wouldn't want to have all kinds of specific exceptions, just a BadRequest seems enough to me.

How should one handle errors in places where there are both full page errors and htmx errors?

**Comment by wimdeblauwe** [comment]
That is a good question, but I have not thought about this, so there is little advice I can give. If you would have a small sample project that shows what you want to do exactly, I can help think and try out some things to see what would work.

**Comment by marccollin** [comment]
you can surely create an restcontrolleradvice and a controlleradvice, put an order on it.
